src/devices/Board/Board.cs

```diff
@@ -0,0 +1,692 @@
+        /// <param name="busId">Bus Id</param>
+        /// <returns>The set of pins for the given I2C bus</returns>
+        public abstract int[] GetDefaultPinAssignmentForI2c(int busId);
+
+        /// <summary>
+        /// Overriden by derived classes: Provides the default pin assignment for the given SPI bus
+        /// </summary>
+        /// <param name="connectionSettings">Connection settings to check</param>
+        /// <returns>The set of pins for the given SPI bus</returns>
+        public abstract int[] GetDefaultPinAssignmentForSpi(SpiConnectionSettings connectionSettings);
+
+        /// <summary>
+        /// Converts a pin number to the logical scheme
+        /// </summary>
+        protected int RemapPin(int pin, PinNumberingScheme providedScheme)
+        {
+            return ConvertPinNumber(pin, providedScheme, PinNumberingScheme.Logical);
+        }
+
+        /// <summary>
+        /// Converts a set of pins to the logical scheme
+        /// </summary>
+        protected int[]? RemapPins(int[] pins, PinNumberingScheme providedScheme)
+        {
+            if (pins == null)
+            {
+                return null;
+            }
+
+            if (providedScheme == PinNumberingScheme.Logical)
+            {
+                return pins;
+            }
+
+            int[] newPins = new int[pins.Length];
+            for (int i = 0; i < pins.Length; i++)
+            {
+                newPins[i] = ConvertPinNumber(pins[i], providedScheme, PinNumberingScheme.Logical);
+            }
+
+            return newPins;
+        }
+
+        private bool UsageCanSharePins(PinUsage usage)
+        {
+            switch (usage)
+            {
+                // Several SPI devices can share the same Pins (because we're allocating devices, not buses)
+                case PinUsage.Spi:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Create an instance of the best possible board abstraction.
+        /// </summary>
+        /// <param name="defaultNumberingScheme">The default pin numbering scheme the board should use</param>
+        /// <returns>A board instance, to be used across the application</returns>
+        /// <exception cref="PlatformNotSupportedException">The board could not be identified</exception>
+        /// <remarks>The detection concept should be refined, but this requires a public detection api</remarks>
+        public static Board Create(PinNumberingScheme defaultNumberingScheme = PinNumberingScheme.Logical)
+        {
+            Board? board = null;
+            try
+            {
+                board = new RaspberryPiBoard(defaultNumberingScheme);
+                board.Initialize();
+            }
+            catch (Exception x) when ((x is NotSupportedException) || (x is IOException))
+            {
+                board?.Dispose();
+                board = null;
+            }
```

Comment:
NIT: All this try and catching seems repetitive, we could extract it to a helper that takes in a Type, or perhaps just a board directly